File: apps/rce-service/src/app/templates/javascript/code.js
{{USER_CODE}}

const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false,
});

rl.on('line', (line) => {
  try {
    const args = line.split(',').map(s => s.trim());
    const res = {{FUNCTION_NAME}};
    console.log(JSON.stringify(res));
  } catch (e) {
    console.error("ERROR:", e && e.stack ? e.stack : e);
  }
});

File: apps/rce-service/src/app/templates/java/code.java

import java.util.*;
import java.util.stream.*;

class Main {
    {{USER_CODE}}
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (sc.hasNextLine()) {
            String line = sc.nextLine().trim();
            if (line.isEmpty()) continue;

            try {
                String[] tokens = Arrays.stream(line.split(","))
                                        .map(String::trim)
                                        .toArray(String[]::new);

                Object ans = {{FUNCTION_NAME}}; 
                System.out.println(ans);

            } catch (Exception e) {
                System.err.println("ERROR: " + e.getMessage());
            }
        }
    }
}
